File: 0070-climbing-stairs/0070-climbing-stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        std::ios_base::sync_with_stdio(false);
        std::cin.tie(nullptr);
        std::cout.tie(nullptr);

        int dp[n + 1];
        if (n == 1) return 1;
        if (n == 2) return 2;
        
        dp[0] = 1, dp[1] = 1;
        for(int i = 2; i <= n; i++) dp[i] = dp[i - 1] + dp[i - 2];

        return dp[n];
    }
};

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution(object):
    def climbStairs(self, n):
        if n == 0:
            return 1
        elif n == 1:
            return 1
        elif n == 2:
            return 2
        else:
            fib1 = 1
            fib2 = 2
            for _ in range(3, n + 1):
                fib = fib1 + fib2
                fib1, fib2 = fib2, fib
            return fib2